### ISSUE TITLE
Multiple URL per block

### ISSUE THREAD

**Issue by osmano807** [rlhk/logseq-url-plus#20]
When editing an existing block with multiple URL, I get strange behaviour. When trying to add a URL+ [title](url) it only adds for one of the URL, even when invoked on each URL, and uses the last URL always.
![Recording 2024-01-01 at 13 49 15](https://github.com/rlhk/logseq-url-plus/assets/321379/dcc74894-8887-4c3c-b855-85836f6e57c9)

**Comment by rlhk** [comment]
Yes, the plugin works on the last token of a block is the expected behavior by design. I'm not sure if what you want could be achieved. Shall take a look when time allows.

**Comment by leojin** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
Luchia777, @YikSanChan, @DevDavid1307, @raulrpearson, @Nemusonaneko, @sanonz, @Samuellyworld, @jkyin, @obiknows, @melvnl, @chrispuska, @tomatowax, @SoufGuy76, @unix, @giacomorebonato, @caaatisgood, @HGWA333, @Rahmteen, @leeseon, @wsadczh

**Comment by rlhk** [comment]
Fixed in [0.3.0](https://github.com/rlhk/logseq-url-plus/releases/tag/0.3.0). Thanks for the recording — it showed the problem exactly. Sorry it sat this long.

My earlier answer was wrong to call "last token" a design decision. It isn't one. It's what falls out of how the plugin read the block — take the whole string, regex the final token off the end — and it got written into the README afterwards as though it had been chosen. Acting on the URL your cursor is on is the reasonable expectation, especially since the plugin advertises multi-term and multiline blocks, which is exactly where it bites.

**What changed**

- Commands act on the **token at your cursor**. In your example, putting the cursor on each URL in turn now formats that URL and leaves the rest of the block alone.
- A block with **exactly one URL** uses it wherever the cursor is, so `see https://example.com for details` no longer fails with `invalid URL "details"`.
- New command **`URL+ All links in block`** converts every bare URL in one pass. URLs that are already markdown links are skipped, and any whose title can't be fetched are left as they were.

With the cursor at the end of a block — where typing a URL then `/` leaves it — the output is byte-identical to 0.2.0, so nothing about the common case moved.

**Two things worth knowing**

- **Put the cursor at the *end* of a URL, not inside it.** Logseq only opens the slash menu when `/` follows a space, so typing that space in the middle of a URL splits the URL before the command ever runs. That's host behaviour I can't work around from the plugin.
- **`URL+ Append Word Definition` always uses the word at your cursor**, never a URL, even in a block that has one. And **`URL+ Inspector ...` still opens on the last token** — it has an editable Token field if you want a different one.

If what you actually wanted was "convert every URL in this block", that's the new command rather than a change to the existing ones — "all of them" has no sensible meaning for the commands that write a single child block or a single `title::` property.

No need to reply if you've long since moved on.
